### PR TITLE
[#3825] Public categories endpoint

### LINKED ISSUE #3825: Retrieving the available forms via the Formulieren API (/api/v2/forms) is slow
### Thema / Theme

API

### Omschrijving / Description

We noticed in OIP 1768 that the /api/v2/forms endpoint is slower than our 5-second timeout, it is now 10-12 seconds on average.

We'll increase our timeout duration as a workaround and the OF testenvironment has quite a few complicated forms, but would it be possible to speed up this endpoint?

The integration in OIP uses Joeri's OF client: https://github.com/open-formulieren/open-forms-client-django 

image

### Added value / Toegevoegde waarde

_No response_

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #3825

### CODE REVIEW

**Comment by sergei-maertens** [comment]
> I feel that maybe this endpoint should also move to the new public folder 

Yep, let's do it if it hasn't been done yet

File: src/openforms/forms/api/viewsets.py [hunk 1]
```diff
@@ -817,6 +817,7 @@ class CategoryViewSet(viewsets.ReadOnlyModelViewSet):
     queryset = Category.objects.all()
     serializer_class = CategorySerializer
     # TODO this permission isn't ideal with its mix of public and write access
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
     permission_classes = [FormAPIPermissions]
```
Comment (sergei-maertens) [inline src/openforms/forms/api/viewsets.py:821]: I think a separate permission class for this that checks if you have explicit `view_category` or `FormAPIPermissions` seems appropriate

I think in DRF these days you can do `Permission1 | Permission2` to do an OR

Comment (SilviaAmAm) [reply src/openforms/forms/api/viewsets.py:821]: But wouldn't this be problematic for the POST actions?

Comment (sergei-maertens) [reply src/openforms/forms/api/viewsets.py:821]: it's a `ReadOnlyModelViewSet` :)

but it's a good idea to add some tests that you get an HTTP 405 on write actions through the API, yes.

Comment (SilviaAmAm) [reply src/openforms/forms/api/viewsets.py:821]: oh oups, silly me :see_no_evil:

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**